### PR TITLE
Vector data cubes

### PR DESCRIPTION
**Pull request by m-mohr**
Minor additions to the API only, Mostly reflecting the changes in https://github.com/stac-extensions/datacube/pull/10 and the addition to the JSON Schema for the new `datacube` subtype.

### CODE REVIEW

**Comment by m-mohr** [comment]
This is ready for final review.

File: openapi.yaml [hunk 1]
```diff
@@ -1375,7 +1376,7 @@ paths:
               example:
                 stac_version: 1.0.0
                 stac_extensions:
-                  - https://stac-extensions.github.io/datacube/v2.0.0/schema.json
+                  - https://stac-extensions.github.io/datacube/v2.2.0/schema.json
```
Comment (dthiex) [inline openapi.yaml:1379]: This link does not work yet.

Comment (m-mohr) [reply openapi.yaml:1379]: Yes, the extension is also waiting for review and release.

File: openapi.yaml [hunk 2]
```diff
@@ -4820,6 +4855,37 @@ components:
...
+      title: Geometry type
+      type: string
+      enum:
+        - Point
+        - MultiPoint
+        - LineString
+        - MultiLineString
+        - Polygon
+        - MultiPolygon
+        - GeometryCollection
```
Comment (soxofaan) [inline openapi.yaml:4868]: Shouldn't there be  a note here (like in other places) that GeometryCollection should be avoided?

**Review by soxofaan — APPROVED** [review]
approve (with minor remark)

File: openapi.yaml (hunk 2 above)
Comment (m-mohr) [reply openapi.yaml:4868]: No, I don't think it makes a lot of sense here. This is backend-oriented documentation. Back-ends can't really "avoid" GeometryCollections, they either support it or not. The data that may consist of GeometryCollections is provided in a different place (processes) by a different audience (end-users) and for this case it's documented in the processes.

**Comment by m-mohr** [comment]
This is also ready for final review. There are some issues with the relevant tooling (redoc), so I can't express everything 1:1 in the API spec. The name for the vector dimension is now "geometries" (plural) in accordance with "bands", okay?

@soxofaan @mkadunc @aljacob @clausmichele @jdries @dthiex @LukeWeidenwalker @pierocampa

**Comment by soxofaan** [comment]
> The name for the vector dimension is now "geometries" (plural) in accordance with "bands", okay?

I don't think that changing to plural "geometries" is really necessary. Apart from "bands", all dimension types and names are/look singular:  "spatial", "temporal", "x", "y", "other".

As another data point here: geopandas uses singular "geometry" as column type and name in data frames, e.g. see https://geopandas.org/en/stable/docs/reference/api/geopandas.GeoDataFrame.html#geopandas.GeoDataFrame

**Review by pierocampa — APPROVED** [review]

**Review by aljacob — APPROVED** [review]

**Review by LukeWeidenwalker — APPROVED** [review]

**Comment by m-mohr** [comment]
> I don't think that changing to plural "geometries" is really necessary.

Are you saying 
>            * `geometries` for dimensions of type `geometries`

should change to
>           * `geometry` for dimensions of type `geometries`

?

**Comment by soxofaan** [comment]
I guess so. 

If I understand correctly, the dimension **type** has already been settled on plural "geometries" in merged PR https://github.com/stac-extensions/datacube/pull/10
so in this PR #441 there is only room to change the dimension **name** to singular "geometry", which would indeed not look very consistent in the recommendation
```
* name `geometry` for dimensions of type `geometries`
```

Sidenote: I noticed that @aljacob made related comment at https://github.com/Open-EO/openeo-processes/pull/382#pullrequestreview-1261722159

**Comment by m-mohr** [comment]
In principle we can still change everything as we just merged the PRs yet, but we have not released anything yet (including the STAC extension). Anyway, I'm merging this now and will then create a separate issue for this. Makes it easier to work on further improvements.

**Pull request merged**